### PR TITLE
Version update mphealth 3.1

### PR DESCRIPTION
**Pull request by Kevcpro**
address https://github.com/OpenLiberty/guides-common/issues/683

### CODE REVIEW

**Review by gkwan-ibm — CHANGES_REQUESTED** [review]
@Kevcpro 
- ** What you’ll learn**
  - [x] need to update "MicroProfile offers checks for both liveness and readiness."
- **Try what you’ll build**
  - [x] "liveness, startup and readiness" -> "startup, liveness, and readiness"
- **Adding health checks to microservices**
  - [x] "The `@Startup` annotation in...", the hotspot `@Startup` highlights incorrectly
  - [x] for startup, `90%` does not match to the SystemStartupCheck code
  - [x] "The `@Liveness` annotation...", the hotsport `@Liveness` highlights incorrectly
  - [x] "The `@Readiness` annotation ... `ApplicationScoped` ...", the hotsports `@Readiness` and `ApplicationScoped` highlight incorrectly

look like hotspots and file references after this section are incorrect. Cannot review.

**Comment by gkwan-ibm** [comment]
hi @Kevcpro 
Would you update both startup classes in the finish, so that the status includes system load data as the liveness check?

**Comment by gkwan-ibm** [comment]
hi @Kevcpro 
Because the testing env may have > 95% cpu usage, the test failed. Let remove the startup tests from the content and junit.

**Review by ryan-storey — APPROVED** [review]

**Comment by ryan-storey** [comment]
System and Inventory Startup and Liveness checks have been modified to pass the tests ran via GitHub actions. Tests now pass consistently. All LGTM!

**Review by pgunapal — APPROVED** [review]
LGTM

File: README.adoc [hunk 1]
```diff
@@ -100,30 +101,37 @@ curl -s http://localhost:9080/inventory/systems | jq
 {: codeblock}
 
 Visit the http://localhost:9080/health URL to see the
-overall health status of the application, as well as the aggregated data of the liveness
+overall health status of the application, as well as the aggregated data of the startup, liveness
```
Comment (dmuelle) [inline README.adoc]: add a comma after liveness

...aggregated data of the startup, liveness, and readiness...

File: README.adoc [hunk 2]
```diff
@@ -156,13 +164,16 @@ A health report will be generated automatically for all services that enable Mic
 [hotspot=mpHealth]`mpHealth` feature has already been enabled for you in the [hotspot]`src/main/liberty/config/server.xml`
 file.
 
-All services must provide an implementation of the `HealthCheck` interface, which will be used to
-verify their health. MicroProfile Health offers health checks for both readiness and liveness.
+All services must provide an implementation of the `HealthCheck` interface, that are used to
```
Comment (dmuelle) [inline README.adoc]: All services must provide an implementation of the `HealthCheck` interface, that are used to
verify their health.
--->
All services must provide an implementation of the `HealthCheck` interface, which is used to
verify their health.

File: README.adoc [hunk 3]
```diff
@@ -176,7 +187,39 @@ include::finish/src/main/liberty/config/server.xml[]
...
+`src/main/java/io/openliberty/guides/system/SystemStartupCheck.java`
+----
+
+SystemStartupCheck.java
+[source, java, linenums, role='code_column tags=SystemStartupCheck hide_tags=copyright']
+----
+include::finish/src/main/java/io/openliberty/guides/system/SystemStartupCheck.java[]
+----
+
+The [hotspot=Startup file=0]`@Startup` annotation indicates that this is a startup health check procedure.
```
Comment (dmuelle) [inline README.adoc]: is the procedure a bean? if so

The [hotspot=Startup file=0]`@Startup` annotation indicates that this bean is a startup health check procedure.

if not a bean- edit as needed to say what "this" is

File: README.adoc [hunk 4]
```diff
@@ -176,7 +187,39 @@ include::finish/src/main/liberty/config/server.xml[]
...
+`src/main/java/io/openliberty/guides/system/SystemLivenessCheck.java`
+----
+
+SystemLivenessCheck.java
+[source, java, linenums, role='code_column tags=SystemLivenessCheck hide_tags=copyright']
+----
+include::finish/src/main/java/io/openliberty/guides/system/SystemLivenessCheck.java[]
+----
+
+The [hotspot=Liveness file=1]`@Liveness` annotation indicates that this is a liveness health check procedure.
```
Comment (dmuelle) [inline README.adoc]: see previous comment re "this"

File: README.adoc [hunk 5]
```diff
@@ -190,8 +233,8 @@ include::finish/src/main/java/io/openliberty/guides/system/SystemReadinessCheck.
 
 // static guide instructions:
 ifndef::cloud-hosted[]
-The [hotspot=Readiness file=0]`@Readiness` annotation indicates that this particular bean is a readiness health check procedure.
-By pairing this annotation with the [hotspot=ApplicationScoped]`ApplicationScoped` context from the Contexts and
+The [hotspot=Readiness file=2]`@Readiness` annotation indicates that this particular bean is a readiness health check procedure.
```
Comment (dmuelle) [inline README.adoc]: don't need "particular" here

File: README.adoc [hunk 6]
```diff
@@ -205,37 +248,50 @@ endpoint receives a request.
 endif::[]
 
 
-The [hotspot=healthCheckResponse]`call()` method is used to return the health status of a particular service.
-In this case, you are simply checking if the server name is [hotspot=defaultServer]`defaultServer` and
-returning [hotspot=HealthCheckResponse-UP]`UP` if it is, and [hotspot=HealthCheckResponse-DOWN]`DOWN` otherwise.  
-Overall, this is a very simple implementation of the [hotspot=healthCheckResponse]`call()`
+The [hotspot=healthCheckResponse file=2]`call()` method is used to return the health status of a particular service.
+In this case, you are simply checking if the server name is [hotspot=defaultServer file=2]`defaultServer` and
```
Comment (dmuelle) [inline README.adoc]: don't need "simply" here

File: README.adoc [hunk 7]
```diff
@@ -205,37 +248,50 @@ endpoint receives a request.
...
+returning [hotspot=HealthCheckResponse-UP file=2]`UP` if it is, and [hotspot=HealthCheckResponse-DOWN file=2]`DOWN` otherwise. 
+Overall, this is a very simple implementation of the [hotspot=healthCheckResponse file=2]`call()`
```
Comment (dmuelle) [inline README.adoc]: suggested edit

Overall, this is a very simple implementation of the [hotspot=healthCheckResponse file=2]`call()`
method. In a real environment, you would want to orchestrate more meaningful
health checks.
--->
This example is a very simple implementation of the [hotspot=healthCheckResponse file=2]`call()`
method. In a real environment, you would orchestrate more meaningful
health checks.

File: README.adoc [hunk 8]
```diff
@@ -247,37 +303,22 @@ InventoryReadinessCheck.java
 include::finish/src/main/java/io/openliberty/guides/inventory/InventoryReadinessCheck.java[]
 ----
 
-In the [hotspot=isHealthy file=0]`isHealthy()` method, 
+In the [hotspot=isHealthy file=2]`isHealthy()` method, 
 you report the `inventory` service as not ready if the service is in maintenance or if its dependant service is unavailable.
 
-For simplicity, the custom [hotspot=3 file=2]`io_openliberty_guides_inventory_inMaintenance`
-MicroProfile Config property, which is defined in the [hotspot file=2]`resources/CustomConfigSource.json`
+For simplicity, the custom [hotspot=3 file=3]`io_openliberty_guides_inventory_inMaintenance`
```
Comment (dmuelle) [inline README.adoc:309]: edit suggestions

For simplicity, the custom [hotspot=3 file=3]`io_openliberty_guides_inventory_inMaintenance`
MicroProfile Config property, that is defined in the [hotspot file=3]`resources/CustomConfigSource.json`
file, is used to indicate whether the service is in maintenance.
--->
For simplicity, the custom [hotspot=3 file=3]`io_openliberty_guides_inventory_inMaintenance`
MicroProfile Config property, which is defined in the [hotspot file=3]`resources/CustomConfigSource.json`
file,  indicates whether the service is in maintenance.

File: README.adoc [hunk 9]
```diff
@@ -454,22 +509,24 @@ You see the following output:
...
 [INFO] Results:
-[INFO]
+[INFO] 
 [INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0
 ----
 
-The warning messages are expected. The first warning results from a request to a service that is under maintenance. This request is made in the `testIfInventoryServiceIsDown()` test from the `InventoryEndpointIT` integration test. The second warning and error results from a request to a bad or an unknown hostname. This request is made in the `testUnknownHost()` test from the `InventoryEndpointIT` integration test.
+The warning messages are expected. The first warning results from a request to a service that is under maintenance. This request is made in the `testHealth()` test from the `InventoryEndpointIT` integration test. The second warning and error results from a request to a bad or an unknown hostname. This request is made in the `testUnknownHost()` test from the `InventoryEndpointIT` integration test.
+
+The tests may fail if your system's CPU or memory is high in usage. The status of the system will be `DOWN` if the CPU usage is over 95%, or the memory usage is over 90%.
```
Comment (dmuelle) [inline README.adoc]: The tests may fail if your system's CPU or memory is high in usage. The status of the system will be `DOWN` if the CPU usage is over 95%, or the memory usage is over 90%.
--->
The tests might fail if your system CPU or memory use is high. The status of the system is `DOWN` if the CPU usage is over 95%, or the memory usage is over 90%.

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review, let me know if you have any questions. Thanks

File: README.adoc (hunk 3 above)
Comment (gkwan-ibm) [reply README.adoc]: I am not sure why before use the word "bean". I assume it implies "bean" = "class". Yes, all 3 classes are similar but for different health checks. So should say "...annotation indicates that this bean..." or "... annotation indicates that this class ..."?

Comment (dmuelle) [reply README.adoc]: If "class" is correct for all 3 instances, I'd recommend that for consistency in all three cases, like:

The [hotspot=Startup file=0]@Startup annotation indicates that this class is a startup health check procedure.

**Review by dmuelle — CHANGES_REQUESTED** [review]

**Review by dmuelle — APPROVED** [review]
LGTM, thanks

**Review by gkwan-ibm — APPROVED** [review]

**Pull request merged**